src/compiler/array.rs

```diff
@@ -224,3 +224,158 @@ pub(crate) fn get(compiler: &mut Compiler, arr: &Symbol, index: &Symbol) -> Symb
+                Instruction::Dup(None),
+                // [i, i]
+                Instruction::MemLoad(Some(length(arr).memory_addr)),
+                // [len, i, i]
+                Instruction::U32CheckedLT,
+                // [i < len, i]
+                Instruction::MemLoad(Some(finished.memory_addr)),
+                // [finished, i < len, i]
+                Instruction::Not,
+                // [!finished, i < len, i]
+                Instruction::And,
+                // [i < len && !finished]
+            ],
+            body: vec![
+                Instruction::Dup(None),
+                // [i, i]
+                Instruction::MemStore(Some(current_element_index.memory_addr)),
+                // [i]
+                Instruction::Dup(None),
+                // [i, i]
+                Instruction::Push(current_element.type_.miden_width()),
+                // [inner_width, i, i]
+                Instruction::U32CheckedMul,
+                // [offset = i * inner_width, i]
+                Instruction::MemLoad(Some(data_ptr(arr).memory_addr)),
+                // [data_ptr, offset, i]
+                Instruction::U32CheckedAdd,
+                // [ptr = data_ptr + offset, i]
+            ]
+            .into_iter()
+            .chain((0..current_element.type_.miden_width()).flat_map(|y| {
+                vec![
+                    Instruction::Dup(None),
+                    // [ptr, ptr, i]
+                    Instruction::Push(y),
+                    // [y, ptr, ptr, i]
+                    Instruction::U32CheckedAdd,
+                    // [ptr + y, ptr, i]
+                    Instruction::MemLoad(None),
+                    // [value, ptr, i]
+                    Instruction::MemStore(Some(current_element.memory_addr + y)),
+                    // [ptr, i]
+                ]
+            }))
+            .chain([Instruction::Drop])
+            .chain(body)
+            .chain([
+                Instruction::Push(1),
+                // [1, i]
+                Instruction::U32CheckedAdd,
+                // [i = i + 1]
+            ])
+            .collect(),
+        },
+        // [i]
+        Instruction::Drop,
+    ]);
+
+    Ok(())
+}
+
+pub(crate) fn includes(compiler: &mut Compiler, arr: &Symbol, el: &Symbol) -> Result<Symbol> {
+    ensure_eq_type!(arr, Type::Array(_));
+    let element_type = element_type(&arr.type_);
+    ensure_eq_type!(el, @element_type);
+
+    let result = compiler
+        .memory
+        .allocate_symbol(Type::PrimitiveType(PrimitiveType::Boolean));
+
+    let index = find_index(compiler, arr, el)?;
+    compiler.memory.read(
+        compiler.instructions,
+        index.memory_addr,
+        index.type_.miden_width(),
+    );
+    // [index]
+    compiler.instructions.extend([
+        Instruction::Push(-1i32 as u32),
+        // [-1, index]
+        Instruction::U32CheckedNeq,
+        // [index != -1]
+        Instruction::MemStore(Some(result.memory_addr)),
+    ]);
```

Comment:
According to [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) it should work similarly to `array.iter().any(|x| x == el)`, no?

Reply:
Yes, except for `NaN`, `indexOf() != -1` and `includes` should work the same. [NaN].includes(NaN) returns true, indexOf will return -1. Rust's `contains` returns false for this, so I don't think it's bad to keep it as `false`.